File: 2021/Python/Day_1.py
def parse_input():
    # Parse data from txt file
    total_arr = []

    with open("Inputs/day_1_input.txt") as input_file:
        for line in input_file:
            total_arr.append(line)

    return total_arr


def part_1():
    depths = parse_input()
    increased = 0

    for i in range(1, len(depths)):
        if int(depths[i]) > int(depths[i - 1]):
            increased += 1

    print("The total increased depths in Part 1 is: " + str(increased))


def part_2():
    depths = parse_input()
    increased = 0
    previous = 0

    for i in range(0, len(depths) - 2, 1):
        slide_window = int(depths[i]) + int(depths[i + 1]) + int(depths[i + 2])
        if previous != 0:
            if slide_window > previous:
                increased += 1

        previous = slide_window

    print("The total increased depths in Part 2 is: " + str(increased))


if __name__ == "__main__":
    part_1()
    part_2()

File: 2021/cpp/Day_1.cpp
#include <fstream>
#include <iostream>
#include <vector>

void part_1();
void part_2();
std::vector<std::string> parse_input();

int main() {
  part_1();
  part_2();
  return 0;
}

void part_1() {
  std::vector<std::string> depths = parse_input();
  int increased = 0;

  for (int i = 1; i < depths.size(); i++) {
    if (stoi(depths[i]) > stoi(depths[i - 1])) {
      increased++;
    }
  }

  std::cout << "The total incrased depths for Part 1 is: " << increased
            << std::endl;
}
void part_2() {
  std::vector<std::string> depths = parse_input();
  int increased = 0;
  int previous = 0;

  for (int i = 0; i < depths.size() - 2; i++) {
    int current = stoi(depths[i]) + stoi(depths[i + 1]) + stoi(depths[i + 2]);

    if (previous != 0) {
      if (current > previous) {
        increased++;
      }
    }
    previous = current;
  }

  std::cout << "The total incrased depths for Part 1 is: " << increased
            << std::endl;
}

std::vector<std::string> parse_input() {
  std::vector<std::string> nums;
  std::ifstream f("Inputs/day_1_input.txt");

  std::string s;

  while (getline(f, s)) {
    nums.push_back(s);
  }

  f.close();

  return nums;
}
